src/sparkdantic/model.py

```diff
@@ -118,29 +156,70 @@ def create_spark_schema(
         override = field_info_extra.get('spark_type')
         field_type = _get_union_type_arg(info.annotation)
 
+        spark_type: Union[str, Dict[str, Any]]
+
         try:
             if _is_base_model(field_type):
-                spark_type = create_spark_schema(field_type, safe_casting, by_alias, mode)
+                spark_type = create_json_spark_schema(field_type, safe_casting, by_alias, mode)
             elif override is not None:
-                if not inspect.isclass(override) or not issubclass(override, spark_types.DataType):
-                    raise TypeError(
-                        '`spark_type` override should be a `pyspark.sql.types.DataType`'
-                    )
-                spark_type = override()
-            elif inspect.isclass(field_type) and issubclass(field_type, spark_types.DataType):
-                spark_type = field_type()
+                if isinstance(override, str):
+                    spark_type = override
```

Comment:
It's possible that a client could provide a bad type name here e.g. `array<int>` and the issue won't be surfaced until the schema is resolved by the SQL parser.

I don't think we should do any validation though. At best we could log a warning, but we'd be maintaining a set of overridable types. @mitchelllisle what do you think?

Reply:
Yeah I think we might need to revisit this to make it a bit more obvious when something is passed in that will cause this. But for now I'm ok leaving it the way it is